### PR TITLE
Add documentation for parallel execution

### PR DESCRIPTION
**Pull request by jqmp**
DO NOT MERGE until parallel execution is ready for release.

### CODE REVIEW

File: docs/concepts.rst [hunk 1]
```diff
@@ -735,6 +737,55 @@ values by "case" instead of by entity.
 Other Features
 --------------
 
+Parallel Execution
```
Comment (namanjain) [inline docs/concepts.rst:740]: How do you feel about adding `.. versionadded:: 0.8.0` for parallel execution?

Comment (namanjain) [inline docs/concepts.rst:740]: Can you also add an entry for this in release notes?

Comment (namanjain) [reply docs/concepts.rst:740]: Oh, now that we release notes, I wonder if we even need any of these `versionadded`. Users can check the release notes for when the feature or concept was released.

Comment (jqmp) [reply docs/concepts.rst:740]: This still sounds like a good idea to me, although I could go either way.

**Pull request merged**